src/plugins/validate-semantic/validators/oas3/refs.js

```diff
@@ -29,3 +30,38 @@ export const validateOAS3RefsForRequestBodiesReferenceRequestBodyPositions = ()
       }, [])
     })
 }
+
+  
+export const validateOAS3RequestBodyRefsReferenceAllowableSchemaPositions = () => sys => {
+  return sys.validateSelectors
+    .allOAS3RequestBodySchemas()
+    .then(nodes => {
+      return nodes.reduce((acc, node) => {
+        const value = node.node
+        const ref = value.$ref
+
+        if(!ref) {
+          return acc
+        }
+
+        const [, refPath = ""] = ref.split("#")
+        const pathArr = refPath.split("/") || []
+        const parentRefKey = pathArr.slice(-2)[0]
+        const targetRefKey = pathArr.slice(-1)[0]
+        if(
+          targetRefKey !== "schema"
+          && parentRefKey !== "schemas"
+          && !ref.startsWith("http://")
+          && !ref.startsWith("https://")
+          && !ref.endsWith(".yaml")
```

Comment:
@ralphdoe a more reliable way to check for local vs external $refs is
```suggestion
          && ref.startsWith("#/")
```
This will exclude all external $refs such as
```
http://...
https://...
path/to/...
./path/to/...
../path/to/...
etc.
```

Reply:
Thanks Helen!